src/components/SingleArticle.jsx

```diff
@@ -1,37 +1,74 @@
 import { useState, useEffect } from "react";
 import { useParams } from "react-router-dom";
-import { getArticleById, getCommentsByArticleId } from "../../utils/api";
+import { getArticleById, getCommentsByArticleId, patchArticle } from "../../utils/api";
 import CommentCard from "./CommentCard";
 
 function SingleArticle() {
   const { article_id } = useParams();
   const [article, setArticle] = useState({});
   const [comments, setComments] = useState([]);
-  const [isLoading, setIsLoading] = useState(true);
+  const [articleLoading, setArticleLoading] = useState(true);
+  const [commentsLoading, setCommentsLoading] = useState(true);
+  const [voteError, setVoteError] = useState(null);
+
+  function upVote(article_id) {
```

Comment:
The upVote and downvote functions are VERY similar - could we streamline this by combining them into one function that takes in article_id and inc_votes, reacting based on if the inc_votes is 1 or -1?